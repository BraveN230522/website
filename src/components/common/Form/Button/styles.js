import styled from '@emotion/styled';
import { css } from '@emotion/core';
import Icon from '@components/common/Icon';

const dynamicVariants = ({ variant, theme, outline }) =>
  ({
    primary: css`
      border: solid 2px ${theme.colors.swiamBlue};
      color: ${outline ? theme.colors.swiamBlue : theme.colors.white};
      background: ${outline ? 'transparent' : theme.colors.swiamBlue};

      &:hover {
        border-color: ${theme.colors.swiamBlue};
        background: ${outline ? theme.colors.swiamBlue : theme.colors.white};
        color: ${outline ? theme.colors.white : theme.colors.swiamBlue};
      }
    `,

    secondary: css`
      border: solid 2px ${theme.colors.white};
      color: ${outline ? theme.colors.white : theme.colors.black};
      background: ${outline ? 'transparent' : theme.colors.white};

      &:hover {
        background: ${outline ? theme.colors.white : theme.colors.black};
        color: ${outline ? theme.colors.black : theme.colors.white};
      }
    `,

    tertiary: css`
      border: solid 2px ${theme.colors.swiamBlue};
      color: ${theme.colors.swiamBlue};
      background: ${theme.colors.white};
      padding: 0 15px;

      &:hover {
        border: solid 2px ${theme.colors.white};
      }
    `,

    dark: css`
      border: solid 2px ${theme.colors.black};
      color: ${outline ? theme.colors.black : theme.colors.black};
      background: ${outline ? 'transparent' : theme.colors.white};

      &:hover {
        border-color: transparent;
        background: ${outline ? theme.colors.black : theme.colors.black};
        color: ${outline ? theme.colors.white : theme.colors.black};
      }
    `,
  }[variant]);

const dynamicSizes = ({ size }) =>
  ({
    sm: css`
      height: 27px;
      letter-spacing: -0.08px;
      font-size: 12px;
      font-weight: 500;
    `,

    md: css`
      height: 45px;
      font-weight: 600;
      font-size: 16px;
    `,
  }[size]);

const ButtonWrapper = styled.button`
  ${dynamicVariants};
  ${dynamicSizes};

  display: inline-flex;
  align-items: center;
  align-items: center;
  justify-content: center;
  transition: all 0.3s ease-in;
  cursor: pointer;
  font-family: ${({ theme }) => theme.fonts.Inter};
  border-radius: 25px;
  user-select: none;
  outline: none;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  min-width: 20px;
  opacity: ${({ disabled }) => (disabled ? 0.6 : 1)};
  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};

  ${({ theme }) =>
    theme.generateBreakpoint(
      768,
      `
      top: 10px;
      margin: 25px auto;
      font-size: 12px;
    `
    )};
  ${({ theme }) =>
    theme.generateBreakpoint(
      375,
      `
      margin: 0 auto;
      margin: 25px auto;
      font-size: 12px;
    `
    )};
  svg {
    margin-right: -6px;
  }

  &:hover {
    path {
      stroke: ${({ theme }) => theme.colors.black};
    }
  }
`;

const RightButtonIcon = styled(Icon)`
  margin-right: 6px;

  path {
    transition: all 0.3s ease-in;
    stroke: ${({ theme }) => theme.colors.white};
  }
`;

const LeftButtonIcon = styled(Icon)`
  margin-left: 6px;

  path {
    transition: all 0.3s ease-in;
    stroke: ${({ theme }) => theme.colors.white};
  }
`;

export { ButtonWrapper, RightButtonIcon, LeftButtonIcon };
