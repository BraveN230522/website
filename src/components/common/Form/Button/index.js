import React from 'react';

import { ButtonWrapper, RightButtonIcon, LeftButtonIcon } from './styles';

const Button = ({
  children,
  rightIcon,
  leftIcon,
  size = 'md',
  variant = 'primary',
  outline = false,
  disabled,
  iconColor,
  className,
  onClick,
  target,
  as,
  href,
}) => {
  return (
    <ButtonWrapper
      size={size}
      as={as}
      target={target}
      href={href}
      variant={variant}
      outline={outline}
      className={className}
      disabled={disabled}
      onClick={onClick}
    >
      {!!rightIcon && <RightButtonIcon color={iconColor} name={rightIcon} />}
      {children}
      {!!leftIcon && <LeftButtonIcon color={iconColor} name={leftIcon} />}
    </ButtonWrapper>
  );
};

export default Button;
