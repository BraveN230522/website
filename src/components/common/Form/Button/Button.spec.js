import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import Button from './index';

/**
 * TODO: When graphql request is configured,
 * this toHaveContent tests will be changed for a mock response
 */
describe('<Button />', () => {
  it('Renders a primary button', () => {
    const { container } = render(<Button variant="primary" />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });

  it('Renders a primary button with an outline style', () => {
    const { container } = render(<Button variant="primary" outline />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });

  it('Renders a secondary button', () => {
    const { container } = render(<Button variant="secondary" />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });

  it('Renders a secondary button with an outline style', () => {
    const { container } = render(<Button variant="secondary" outline />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });
});
