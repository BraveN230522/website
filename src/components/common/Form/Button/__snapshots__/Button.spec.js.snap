// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Renders a primary button 1`] = `
.emotion-0 {
  border: solid 2px #2891ff;
  color: #ffffff;
  background: #2891ff;
  height: 45px;
  font-weight: 600;
  font-size: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
  cursor: pointer;
  font-family: Inter,sans-serif;
  border-radius: 25px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 103px;
}

.emotion-0:hover {
  border-color: #2891ff;
  background: #ffffff;
  color: #2891ff;
}

.emotion-0 svg {
  margin-right: 6px;
}

.emotion-0:hover path {
  stroke: #000000;
}

<div>
  <button
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`<Button /> Renders a primary button with an outline style 1`] = `
.emotion-0 {
  border: solid 2px #2891ff;
  color: #2891ff;
  background: transparent;
  height: 45px;
  font-weight: 600;
  font-size: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
  cursor: pointer;
  font-family: Inter,sans-serif;
  border-radius: 25px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 103px;
}

.emotion-0:hover {
  border-color: #2891ff;
  background: #2891ff;
  color: #ffffff;
}

.emotion-0 svg {
  margin-right: 6px;
}

.emotion-0:hover path {
  stroke: #000000;
}

<div>
  <button
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`<Button /> Renders a secondary button 1`] = `
.emotion-0 {
  border: solid 2px #ffffff;
  color: #000000;
  background: #ffffff;
  height: 45px;
  font-weight: 600;
  font-size: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
  cursor: pointer;
  font-family: Inter,sans-serif;
  border-radius: 25px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 103px;
}

.emotion-0:hover {
  background: #000000;
  color: #ffffff;
}

.emotion-0 svg {
  margin-right: 6px;
}

.emotion-0:hover path {
  stroke: #000000;
}

<div>
  <button
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`<Button /> Renders a secondary button with an outline style 1`] = `
.emotion-0 {
  border: solid 2px #ffffff;
  color: #ffffff;
  background: transparent;
  height: 45px;
  font-weight: 600;
  font-size: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
  cursor: pointer;
  font-family: Inter,sans-serif;
  border-radius: 25px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 103px;
}

.emotion-0:hover {
  background: #ffffff;
  color: #000000;
}

.emotion-0 svg {
  margin-right: 6px;
}

.emotion-0:hover path {
  stroke: #000000;
}

<div>
  <button
    class="emotion-0 emotion-1"
  />
</div>
`;
